### ISSUE TITLE
Connection error using mongoose v5.10.0 with mongodb-memory-server

### ISSUE THREAD

**Issue by DkITJamesFarrell** [Automattic/mongoose#9348]
**Do you want to request a *feature* or report a *bug*?**

Bug - an issue that I'm having with the latest version of mongoose

**What is the current behavior?**

Failing unit tests when running mongoose v5.10.0 with mongodb-memory-server on multiple test files. Receiving error: *MongooseError: Can't call `openUri()` on an active connection with different connection strings. Make sure you aren't calling `mongoose.connect()` multiple times.* 

I originally thought that this was because I was running the tests in parallel using mocha's new parallel execution feature, but the problem persists when I turn parallel execution off.

**If the current behavior is a bug, please provide the steps to reproduce.**

Example code - each test file uses a similar structure

```
import { describe, it } from 'mocha';
import mongoose from 'mongoose';

import { MongoMemoryServer } from 'mongodb-memory-server';
let mongoServer: MongoMemoryServer;

describe('example test', () => {
  before(async () => {
    mongoServer = new MongoMemoryServer();

    const mongoUri = await mongoServer.getUri();
    const opts = {
      useCreateIndex: true,
      useFindAndModify: false,
      useNewUrlParser: true,
      useUnifiedTopology: true,
    };

    await mongoose.connect(mongoUri, opts);
  });

  after(async () => {
    await mongoose.disconnect();
    await mongoServer.stop();
  });

  it(async () => {
    // Some test code here
  });
});
```

**What is the expected behavior?**

All tests should pass without connection errors, as they do with mongoose v5.9.29

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

node v14.8.0
mongoose v5.10.0
mongodb v4.4
mongodb-memory-server v6.6.4

**Comment by vkarpov15** [comment]
Are you using `--watch` or some other mechanism to re-run the tests automatically?

**Comment by DkITJamesFarrell** [comment]
No, but I've figured the problem out -

A mongoose connection is created within an 'app' class which is imported at the top of each test. I fixed my problem by calling mongoose.disconnect() (to disconnect from the app instance) before calling mongoose.connect() (to connect to the memory server instance). 

This has been addressed in issue #9203
